### ISSUE TITLE
Verify if MariaDB Docker image automatically runs mysql_upgrade

### ISSUE THREAD

**Issue by Vaalyn** [AzuraCast/docker-azuracast-db#3]
While going through the issue list of AzuraCast I researched a bit more about the error shown in https://github.com/AzuraCast/AzuraCast/issues/3629 and I think this has probably something to do with the image not running the `mysql_upgrade` command.

I wasn't able to find reliable information if newer version of the official Docker image are doing this or not.
See https://github.com/MariaDB/mariadb-docker/issues/350

This should be investigated.

**Comment by grooverdan** [comment]
Yes, it does look like the sort of issue you'd run into without mysql_upgrade being run. I'll be looking how to resolve this upstream.

**Comment by SlvrEagle23** [comment]
@Vaalyn I've investigated and `mysql_upgrade` (or its MariaDB counterpart in newer versions, `mariadb-upgrade`) is _not_ being run by default at any step in the bootstrapping process for the database.

I've created a little helper script that can run the upgrade steps necessary. There's one problem with the whole arrangement: it needs the root password in order to manipulate the `mysql` database. We generate a random root password and echo it out on the initial startup, but then we don't keep the actual root password anywhere (as, at least until now, we don't need it for anything). Thus, this script needs to temporarily boot MariaDB in "rescue mode", then do the cleanup, then spin down that instance in favor of the regular database connection.

Basically, the upgrade process would look like:

```bash
docker-compose down
docker-compose run --rm mariadb db_upgrade
docker-compose up -d
```

Everything in the upgrade process actually runs fine, but the last step in that script is to `FLUSH PRIVILEGES`, which wakes mariadb up from "rescue mode" and makes it stop and go "hey, wait a minute...", which errors out the very last command. This error isn't actually a problem, as the upgrade goes through without an issue, but you know how it goes with errors-that-aren't-errors being displayed; they'll always wind up being GitHub issues.

I think for now, having this script available to direct some users to if we suspect their errors are caused by upgrading from an older MariaDB version would be helpful, but I wouldn't necessarily want to incorporate it into the `./docker.sh update` process, partly because of the volatility of the "rescue mode" but also because in 99% of updates it wouldn't actually do anything but make the update process a little longer and more confusing.

**Comment by Vaalyn** [comment]
@SlvrEagle23 I think for now it's sufficient having a script on standby that we can direct users to if necessary. That should make it much easier to help users encountering problems related to this.

For the future I think we should rework how we handle the MariaDB root password thing.
In general I think it would be beneficial to generate a random root password and saving it to the env vars since that would enable us (and in special circumstances users too) to do things in the DB that require the root user.

In general I think if we do that for the root password we could also do that for the general db user password too. Although the db in Docker is not accessible from outside the Docker network I think it's a bad thing to have a fixed password for that.

**Comment by SlvrEagle23** [comment]
@Vaalyn Agreed about the root password, but it's worth pointing out we actually _do_ already do that for the main `azuracast` password. A few folks brought up that the fixed password bothered them, even if the connection wasn't public, so now during the initial setup process we generate a random password and replace the static one with it in the environment file just before running the initial database setup. :)

**Comment by Vaalyn** [comment]
@SlvrEagle23 thanks for the link to the code where this is generated. I'm also totally fine with having it be generic for dev environments.
